src/analysis/retail/monk/mistweaver/modules/spells/EnergizingBrew.tsx

```diff
@@ -0,0 +1,68 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_MONK } from 'common/TALENTS';
+import { TooltipElement } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { calculateEffectiveResourceRestored } from 'parser/core/EventCalculateLib';
+import Events, { ApplyBuffEvent, ResourceChangeEvent } from 'parser/core/Events';
+import ItemManaGained from 'parser/ui/ItemManaGained';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import { getManaTeaChannelDuration } from '../../normalizers/CastLinkNormalizer';
+
+const MANA_INCREASE = 0.2;
+
+class EnergizingBrew extends Analyzer {
+  totalRestored: number = 0;
+  timeSaved: number = 0;
+  totalTeas: number = 0;
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(
+      Events.resourcechange.by(SELECTED_PLAYER).spell(SPELLS.MANA_TEA_CAST),
+      this.onManaRestore,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.MANA_TEA_BUFF),
+      this.onApplyBuff,
+    );
+  }
+
+  onManaRestore(event: ResourceChangeEvent) {
+    this.totalRestored += calculateEffectiveResourceRestored(event, MANA_INCREASE);
+  }
+
+  onApplyBuff(event: ApplyBuffEvent) {
+    // channel duration is equal to time saved because its 50% faster
+    this.timeSaved += getManaTeaChannelDuration(event) || 0;
```

Comment:
Could probably do the same thing here that was just done in ManaTea for prepull channels. If we're going to record a 0 duration channel to time saved we shouldn't add 1 to total teas it should probably just return

Reply:
I also don't know how I feel about this for channels <= the gcd. Since you're locked into the gcd regardless you're not really 'saving time' and more so consuming additional stacks. But I guess that's semantics